bimmer_connected/state.py

```diff
@@ -31,6 +31,16 @@ class LockState(Enum):
     UNLOCKED = 'UNLOCKED'
 
 
+class UpdateReason(Enum):
+    """Possible reasons for an update from the vehicle to the server."""
+    VEHICLE_SECURED = 'VEHICLE_SECURED'
+    DOORSTATECHANGED = 'DOORSTATECHANGED'
+    VEHCSHUTDOWN = 'VEHCSHUTDOWN'
+    VEHCSHUTDOWN_SECURED = 'VEHCSHUTDOWN_SECURED'
+    CHARGINGSTARTED = 'CHARGINGSTARTED'
+    ERROR = 'Error'
+
```

Comment:
I guess the list is a bit longer, especially for electric cars. What happens if you receive an unknown update reason?

Reply:
Yes, this is also something that worries me. On the one hand I would like to have a complete list of all values, so that the users of the library can use them to check for certain reasons.

On the other have: If we're missing a value here, the program will raise an exception and that might break in situations where nobody really cares if the value is not parsed correctly.

I did not find a way to move all unknown data into a separate state `UNKNOWN` and basically ignore it.